actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -59,7 +59,7 @@ def check_path_parameters!
       path_parameters.each do |key, value|
         next unless value.respond_to?(:valid_encoding?)
         unless value.valid_encoding?
-          raise ActionController::BadRequest, "Invalid parameter: #{key} => #{value}"
```

Comment:
@rafaelfranca I've updated my comment above, with some proposals. I believe this is something that we could and *should* fix, at least in `master`.

WDYT?